src/components/modals/WcSmartContractModal.tsx

```diff
@@ -193,3 +192,41 @@ const getStyles = cacheStyles((theme: Theme) => ({
     paddingVertical: theme.rem(1)
   }
 }))
+
+async function wcRequestResponse(wallet: EdgeCurrencyWallet, uri: string, approve: boolean, payload: WcRpcPayload): Promise<void> {
+  if (approve) {
```

Comment:
I know this is a copy-paste from the currency plugin, but I would do it `if (!approve) { reject(); return; }` and save a level of indentation for the whole rest of the function. I don't like giant `if` blocks where you have to scroll to see the `else` clause - it's very hard to read.